platforms/jvm/toolchains-jvm/src/testFixtures/groovy/org/gradle/jvm/toolchain/JdkRepository.groovy

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.jvm.toolchain
+
+import org.apache.commons.compress.archivers.ArchiveOutputStream
+import org.apache.commons.compress.archivers.zip.ZipArchiveEntry
+import org.apache.commons.compress.archivers.zip.ZipArchiveOutputStream
+import org.apache.commons.compress.utils.IOUtils
+import org.gradle.api.JavaVersion
+import org.gradle.integtests.fixtures.AvailableJavaHomes
+import org.gradle.internal.jvm.Jvm
+import org.gradle.internal.nativeintegration.services.FileSystems
+import org.gradle.test.fixtures.server.http.BlockingHttpServer
+import org.gradle.util.internal.RelativePathUtil
+
+import java.nio.file.Files
+
+
+class JdkRepository {
+
+    private BlockingHttpServer server
+
+    private Jvm jdk
+
+    private String archiveName
+    private File archiveFile
+
+    JdkRepository(JavaVersion javaVersion) {
+        this(Objects.requireNonNull(AvailableJavaHomes.getJdk(javaVersion)), "jdk.zip")
+    }
+
+    JdkRepository(Jvm jdk, String archiveName) {
+        this.jdk = jdk
+        this.archiveName = archiveName
+
+        archiveFile = zip(jdk.javaHome, File.createTempFile(archiveName, ".tmp").with {deleteOnExit(); it })
```

Comment:
❌ This won't be reused as the `JdkRepository` instance will be recreated for each test. Let's have a chat about it.

Reply:
There is a single instance created per `Specification` (i.e. **not** per test method). Did you mean to cast the net for reuse even wider?

Reply:
I see now that you made it static in specs. That works too.